src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -488,6 +488,36 @@ public static ProtoCore.AssociativeGraph.GraphNode GetFirstSSAGraphnode(int inde
             return nodesInScope[indexOfDirtyNode + 1];
         }
 
+       
+
+        /// <summary>
+        /// Marks a graphnode dirty and returns the graphnode
+        /// </summary>
+        /// <param name="runtimeCore"></param>
+        /// <param name="astID"></param>
+        /// <returns></returns>
+        public static AssociativeGraph.GraphNode MarkGraphNodeDirty(RuntimeCore runtimeCore, int astID)
+        {
+            Executable exe = runtimeCore.DSExecutable;
+            List<AssociativeGraph.GraphNode> nodesInScope = 
+                exe.instrStreamList[0].dependencyGraph.GetGraphNodesAtScope(Constants.kInvalidIndex, Constants.kGlobalScope);
+            foreach (var gnode in nodesInScope)
+            {
+                if (gnode.isActive && gnode.OriginalAstID == astID)
+                {
+                    gnode.isDirty = true;
+                    gnode.isActive = true;
+
+                    if (gnode.updateBlock.updateStartPC != Constants.kInvalidIndex)
+                    {
+                        gnode.updateBlock.startpc = gnode.updateBlock.updateStartPC;
+                    }
+                    return gnode;
+                }
+            }
+            return null;
+        }
+
```

Comment:
I cant use that just yet, MarkGraphNodesDirty marks all nodes regardless of scope. MarkGraphNode will only handle one node and only on the global scope.
I will combine them to one function, but for now, it is risky to change because it can destabilize the live execution behavior:
Added in: